pkg/policy/api/rule_validation.go

```diff
@@ -150,12 +150,24 @@ func (e *EgressRule) sanitize() error {
 // TODO we need to add support to check
 // wildcard and prefix/suffix later on.
 func (kr *PortRuleKafka) Sanitize() error {
+	if (len(kr.APIKey) > 0) && (len(kr.Role) > 0) {
+		return fmt.Errorf("Cannot set both Role:%q and APIKey :%q together", kr.Role, kr.APIKey)
+	}
+
 	if len(kr.APIKey) > 0 {
 		n, ok := KafkaAPIKeyMap[strings.ToLower(kr.APIKey)]
 		if !ok {
-			return fmt.Errorf("invalid Kafka APIKey :%q", kr.APIKey)
+			return fmt.Errorf("invalid Kafka APIKey :%q len(APIKey):%d\n", kr.APIKey, len(kr.APIKey))
```

Comment:
Why is the len needed in the error message?

Reply:
left over of debugging. Will remove it

Reply:
done